agent-inject/agent/annotations.go

```diff
@@ -156,6 +156,17 @@ const (
 	// AnnotationPreserveSecretCase if enabled will preserve the case of secret name
 	// by default the name is converted to lower case.
 	AnnotationPreserveSecretCase = "vault.hashicorp.com/preserve-secret-case"
+
+	// AnnotationAgentCacheEnable if enabled will configure the sidecar container
+	// to enable agent caching
+	AnnotationAgentCacheEnable = "vault.hashicorp.com/agent-cache-enable"
+
+	// AnnotationAgentCacheUseAutoAuthToken configures the agent cache to use the
+	// auto auth token or not. Can be set to "force" to force usage of the auto-auth token
+	AnnotationAgentCacheUseAutoAuthToken = "vault.hashicorp.com/agent-cache-use-auth-auth-token"
+
+	// AnnotationAgentCacheListenerAddress configures the address the agent cache should listen on
+	AnnotationAgentCacheListenerAddress = "vault.hashicorp.com/agent-cache-listener-address"
```

Comment:
Maybe this shouldn't be configurable but always be `127.0.0.1`.  This really shouldn't be listening to anything other than localhost.  Especially since the `use-auth-auth-token` parameter is configurable, unauthenticated clients could retrieve secrets from Vault if misconfigured.

Reply:
I was thinking of allowing the user to change the port. Should I change this into a port configuration then?

Reply:
What is your thoughts with enabling the listener?  If other containers in the pod want to use the vault agent as a proxy?

Reply:
I am not sure what you mean by this question. Do you mean what are my thoughts on allowing the listener to listen to IP addresses other than `127.0.0.1`? I think you're right that it is a bad idea.

Reply:
Just trying to understand the feature more in general.  We definitely don't want to bind to anything other than localhost, port configuration could be useful.

Reply:
I see. I think this is a very useful feature. Other than just proxying, the vault agent cache also takes care of renewing secrets. I have some applications that basically "outsources" the renewal duty to Vault agent.

Reply:
Agreed with it being useful.  I think turning this into a port configurable makes more sense.

Reply:
OK. I will make the change in a bit.

Reply:
Done. See https://github.com/hashicorp/vault-k8s/pull/132/commits/f0dc6868eb5e7f120496cfb43526f861139aee1f